### PR TITLE
Marcus/implement lesson

### LINKED ISSUE #162: Introduce lesson class

### PR DESCRIPTION
**Pull request by MarcTzh**
someone send help

**completed:**
testing for most classes
created Date and Time class

**yet to do:**
LessonCommandParserTest
TypicalLessons
Exception for when LessonCommand has some empty arguments
Exception for when startTime/startDate <= endTime/endDate

KIV:
Beining's idea of introducing module

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ PlaNus is a desktop app for **managing tasks, optimized for use via a Command Li
   * [Show all commands : `help`](#show-all-commands--help)
   * [List all tasks : `list`](#list-all-tasks--list)
   * [Add a task: `add`](#add-a-task-add)
+  * [Add a lesson: `add`](#add-a-lesson-lesson)
```
Comment (BobbyZhouZijian) [inline docs/UserGuide.md]: should be `lesson` right

File: src/main/java/seedu/address/logic/commands/LessonCommand.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        requireNonNull(lesson);
+        this.lesson = lesson;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        ArrayList<Task> tasksToAdd = lesson.createRecurringTasks();
+        for (Task taskToAdd: tasksToAdd) {
+            System.out.println("date: " + taskToAdd.getDateTime());
```
Comment (BobbyZhouZijian) [inline src/main/java/seedu/address/logic/commands/LessonCommand.java]: Should remove this

File: src/main/java/seedu/address/model/lesson/Time.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
+package seedu.address.model.lesson;
```
Comment (BobbyZhouZijian) [inline src/main/java/seedu/address/model/lesson/Time.java:1]: Actually, I think we can make the Time, Date and DateTime class inherit or implement a common interface. Maybe I'll refactor this later.

**Review by BobbyZhouZijian — COMMENTED** [review]
Some nits

File: docs/UserGuide.md [hunk 4]
```diff
@@ -79,6 +80,23 @@ Examples:
...
+### Add a lesson: `lesson`
+
+Adds a lesson to PlaNus.
+
+Format: `lesson title:TITLE [desc:DESCRIPTION] day:DAY from:TIME to:TIME start:DATE end:DATE`
+
+* Adds multiple recurring tasks of  `type:lesson` to PlaNus, starting from the date specified in `start:DATE` to the
+ date specified in `end:DATE`, on the day specified in `day:DAY` from the time specified in `from:TIME` to the time
+ specified in `to:TIME`.
+* The format of day must be as follows (case-insensitive):
```
Comment (GabriellaTeh) [inline docs/UserGuide.md:93]: Not sure if it is just me but i usually write thursday as 'thurs' in short from but it must be 'thu' for the input. Maybe can make it clearer?

**Comment by GabriellaTeh** [comment]
Also currently the message says that the lesson is added when the end date can be set before the start date, but it does not show up on the list. Should change the output message

**Review by BobbyZhouZijian — APPROVED** [review]
Approved the PR so that others can work on new features. Improvements have been added as issues.

**Pull request merged**